### PR TITLE
Move PDOS viewer in a separate class

### PR DESCRIPTION
**Pull request by yakutovicha**
ToDo:

- [x] Merge `utils/pdos_postprocess.py` into `pdos.py`. Remove legacy and unused functions.
- [x] Agree on the outlook.
- [x] Agree what to do with the legacy stuff.

### CODE REVIEW

**Comment by yakutovicha** [comment]
@eimrek do you mind giving this new widget a quick test (and, optionally, a quick review)?

**Comment by yakutovicha** [comment]
@cpignedoli shall we remove legacy functions from the pdos?

File: surfaces_tools/widgets/pdos.py [hunk 1]
```diff
@@ -0,0 +1,727 @@
...
+    fermi = 0.5 * (data[n_el - 1, 1] + data[n_el, 1])
+
+    out_data = np.zeros((data.shape[0], 2))
+    out_data[:, 0] = (data[:, 1] - fermi) * HART_2_EV  # energy
+    out_data[:, 1] = np.sum(data[:, 3:], axis=1)  # "contracted pdos"
+    return out_data, kind
+
+
+def process_pdos_files(scf_calc, newversion=True):
+    if newversion:
```
Comment (cpignedoli) [inline surfaces_tools/widgets/pdos.py:44]: this refers to AiiDA <2.0

File: surfaces_tools/widgets/pdos.py [hunk 2]
```diff
@@ -0,0 +1,727 @@
...
+            break
+
+    dos = {
+        "mol": [None] * nspin,
+    }
+
+    for file in sorted(retr_files):
+        if file.endswith(".pdos"):
+            with retr_folder.open(file) as fhandle:
+                try:
```
Comment (cpignedoli) [inline surfaces_tools/widgets/pdos.py:64]: this type of try/except taht appears not only here had to be introduced with AiiDA 2.0. We still have many data requiring the older open procedure so for the moment we have to keep both

**Review by cpignedoli — APPROVED** [review]
I tested on my account everything seem to work

**Comment by yakutovicha** [comment]
At this point, I think we can merge.

**Pull request merged**